pkg/plugins/runtime/k8s/metadata/annotations.go

```diff
@@ -106,7 +106,7 @@ const (
 	// KumaMetricsPrometheusAggregateEnabled allows to specify if we want to enable specific scraping, default: true
 	KumaMetricsPrometheusAggregateEnabled = "prometheus.metrics.kuma.io/aggregate-%s-enabled"
 	// KumaMetricsPrometheusAggregatePattern allows to retrieve all the apps for which need to get port/path configuration
-	KumaMetricsPrometheusAggregatePattern = "^prometheus.metrics.kuma.io/aggregate-([a-zA-Z0-9-]+)-(port|path|enabled)$"
+	KumaMetricsPrometheusAggregatePattern = "^prometheus\\.metrics\\.kuma\\.io/aggregate-([a-zA-Z0-9-]+)-(port|path|enabled)$"
```

Comment:
why do we need to escape like this?

Reply:
As it is a regular expression, dots must be escaped, I don’t remember which codeql rule was saying that.
I can rollback this fix to what codeql says, if you want ?

Reply:
Should the `regexp.MustCompile` directly be applied on the pattern so it is ready to be used, by the way ?

Reply:
So I think we're missing coverage somewhere if this was working before escaping, or we have `regexp.QuoteMeta` somewhere down the line so golang escapes it. Let me check.

Reply:
We have `MustCompile` here - https://github.com/kumahq/kuma/blob/67c3ef36a0e9336eb4082d39f3f2376cc8bcaf22/pkg/plugins/runtime/k8s/controllers/pod_converter.go#L24 . So I think it works because `.` will match any character - including `.` 😄. I think the escaped form is the correct one but maybe @lukidzi can confirm.

Reply:
Looks like we should escape dots. Good catch, thanks for fixing this @mmorel-35 🎉